File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/AttributeSetInstanceExtensionFieldStateEventIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\FlattenedDtoInterface;

class AttributeSetInstanceExtensionFieldStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'groupId' => 'string',
            'index' => 'string',
            'attributeSetInstanceExtensionFieldGroupVersion' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var AttributeSetInstanceExtensionFieldStateEventId
     */
    private $value;

    /**
     * @param AttributeSetInstanceExtensionFieldStateEventId $value
     */
    public function __construct(AttributeSetInstanceExtensionFieldStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new AttributeSetInstanceExtensionFieldStateEventId();
        }
    }

    /**
     * @return AttributeSetInstanceExtensionFieldStateEventId
     */
    public function toAttributeSetInstanceExtensionFieldStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getGroupId()
    {
        return $this->value->getGroupId();
    }

    /**
     * @param string $groupId
     */
    public function setGroupId($groupId)
    {
        $this->value->setGroupId($groupId);
    }

    /**
     * @return string
     */
    public function getIndex()
    {
        return $this->value->getIndex();
    }

    /**
     * @param string $index
     */
    public function setIndex($index)
    {
        $this->value->setIndex($index);
    }

    /**
     * @return Long
     */
    public function getAttributeSetInstanceExtensionFieldGroupVersion()
    {
        return $this->value->getAttributeSetInstanceExtensionFieldGroupVersion();
    }

    /**
     * @param Long $attributeSetInstanceExtensionFieldGroupVersion
     */
    public function setAttributeSetInstanceExtensionFieldGroupVersion($attributeSetInstanceExtensionFieldGroupVersion)
    {
        $this->value->setAttributeSetInstanceExtensionFieldGroupVersion($attributeSetInstanceExtensionFieldGroupVersion);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getGroupId(),
            $this->getIndex(),
            $this->getAttributeSetInstanceExtensionFieldGroupVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return AttributeSetInstanceExtensionFieldStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setGroupId($pValues[0]);
        $this->setIndex($pValues[1]);
        $this->setAttributeSetInstanceExtensionFieldGroupVersion($pValues[2]);
        return $this;
    }

}

File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/AttributeSetInstanceExtensionFieldStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class AttributeSetInstanceExtensionFieldStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "groupId",
            "index",
            "attributeSetInstanceExtensionFieldGroupVersion",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private AttributeSetInstanceExtensionFieldStateEventId value;

    public AttributeSetInstanceExtensionFieldStateEventIdFlattenedDto()
    {
        this(new AttributeSetInstanceExtensionFieldStateEventId());
    }

    public AttributeSetInstanceExtensionFieldStateEventIdFlattenedDto(AttributeSetInstanceExtensionFieldStateEventId value)
    {
        this.value = value;
    }

    public AttributeSetInstanceExtensionFieldStateEventId toAttributeSetInstanceExtensionFieldStateEventId()
    {
        return this.value;
    }


    public String getGroupId()
    {
        return this.value.getGroupId();
    }

    public void setGroupId(String groupId)
    {
        this.value.setGroupId(groupId);
    }

    public String getIndex()
    {
        return this.value.getIndex();
    }

    public void setIndex(String index)
    {
        this.value.setIndex(index);
    }

    public Long getAttributeSetInstanceExtensionFieldGroupVersion()
    {
        return this.value.getAttributeSetInstanceExtensionFieldGroupVersion();
    }

    public void setAttributeSetInstanceExtensionFieldGroupVersion(Long attributeSetInstanceExtensionFieldGroupVersion)
    {
        this.value.setAttributeSetInstanceExtensionFieldGroupVersion(attributeSetInstanceExtensionFieldGroupVersion);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        AttributeSetInstanceExtensionFieldStateEventIdFlattenedDto other = (AttributeSetInstanceExtensionFieldStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

